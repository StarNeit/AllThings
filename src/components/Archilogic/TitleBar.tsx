import {
  TitleBarBackButton,
  TitleBarButton,
  TitleBarButtonGroup,
  TitleBarLabel,
} from 'components/TitleBar' // tslint:disable-line:no-implicit-dependencies
// tslint:disable-next-line:no-implicit-dependencies
import Localized from 'containers/Localized'
import { css } from 'glamor'
import React from 'react'
import { MicroApps } from 'enums'

interface ITitleBarProps {
  readonly detail?: string
  readonly label: object
  readonly modelDisplayName?: string
  readonly onClick: (e: React.MouseEvent) => void
}

export default function TitleBar({
  detail,
  label,
  modelDisplayName,
  onClick,
}: ITitleBarProps): JSX.Element {
  const handleClick = (e: React.MouseEvent) => {
    e.preventDefault()
    onClick(e)
    return false
  }
  return detail ? (
    <div {...css({ display: 'flex', width: '100%' })}>
      <TitleBarBackButton
        onClick={handleClick}
        data-e2e="archilogic-show-back"
      />
      <TitleBarButtonGroup direction="right">
        <TitleBarButton label={modelDisplayName} />
      </TitleBarButtonGroup>
    </div>
  ) : (
    <TitleBarLabel microAppType={MicroApps.ARCHILOGIC}>
      <Localized messages={label} />
    </TitleBarLabel>
  )
}
