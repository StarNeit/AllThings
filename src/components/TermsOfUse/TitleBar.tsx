import React from 'react'
import { View, Icon, Text } from '@allthings/elements'
import { ColorPalette } from '@allthings/colors'
import { css } from 'glamor'
import NoOp from 'utils/noop'
import { MARGIN } from './'
import { calculateHeightOfNativeApp } from 'utils/guessBrowser'

const calculateHeight = (isNativeApp: boolean) =>
  isNativeApp ? calculateHeightOfNativeApp() : 0

const styles = (loggedIn: boolean, isNativeApp: boolean) => ({
  titleBar: css({
    background: ColorPalette.lightBlueIntense,
    borderRadius: loggedIn ? 0 : `${MARGIN / 2}px ${MARGIN / 2}px 0 0`,
    cursor: loggedIn ? 'auto' : 'pointer',
    height: `${MARGIN * (loggedIn ? 5 : 4) + calculateHeight(isNativeApp)}px`,
  }),
  titleBarItems: css({
    margin: `${MARGIN}px`,
    paddingTop: `${calculateHeight(isNativeApp)}px`,
  }),
})

interface IProps {
  isNativeApp?: boolean
  loggedIn?: boolean
  onClick: OnClick
  title: string
}

export const TitleBar = ({
  isNativeApp = false,
  loggedIn = false,
  onClick = NoOp,
  title,
}: IProps) => {
  const { titleBar, titleBarItems } = styles(loggedIn, isNativeApp)
  return (
    <View
      alignH="space-between"
      alignV="center"
      direction="row"
      onClick={loggedIn ? NoOp : onClick}
      {...titleBar}
    >
      <Text color={ColorPalette.white} size="l" strong {...titleBarItems}>
        {title}
      </Text>
      {!loggedIn && (
        <Icon
          color={ColorPalette.white}
          name="remove-light-filled"
          size="xs"
          {...titleBarItems}
        />
      )}
    </View>
  )
}
