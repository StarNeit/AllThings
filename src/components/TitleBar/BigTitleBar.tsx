import { ColorPalette } from '@allthings/colors'
import { Button, SquareIconButton, Text, View } from '@allthings/elements'
import { createMQ } from '@allthings/elements/Responsive'
import ScalableTextContainer from 'components/ScalableTextContainer'
import { css } from 'glamor'
import React, { ReactNode } from 'react'

export interface IBigTitleBarProps {
  title: ReactNode
  subTitle?: ReactNode
  buttonText?: ReactNode
  onButtonClick?: () => void
  isTwoColumnLayout?: boolean
  showBackButton?: boolean
  onBackButtonClick?: () => void
}

const titleStyle = css({
  fontSize: 42,
  fontWeight: 'bold',
  lineHeight: '1em',
  marginBottom: 7,
  letterSpacing: -2,
  wordWrap: 'break-word',
})

const BigTitleBar = ({
  title,
  buttonText,
  onButtonClick,
  subTitle,
  isTwoColumnLayout = false,
  showBackButton = false,
  onBackButtonClick,
  ...props
}: IBigTitleBarProps) => (
  <View direction="row" {...props}>
    {showBackButton && (
      <SquareIconButton
        icon="arrow-left-filled"
        iconSize="s"
        onClick={onBackButtonClick}
        {...css({
          paddingTop: 20,
          paddingLeft: 10,
          marginRight: -3,
          transform: 'translateY(13px)',
        })}
      />
    )}
    <View
      flex="grow"
      {...css({
        minWidth: 0,
        width: '100%',
        padding: '20px 15px 0px',
        [createMQ('desktop', 'tablet')]: !isTwoColumnLayout && {
          padding: '20px 0 0 0',
        },
      })}
    >
      <View {...titleStyle}>
        <ScalableTextContainer>{title}</ScalableTextContainer>
      </View>
      {subTitle && (
        <Text color={ColorPalette.grey}>
          <ScalableTextContainer>{subTitle}</ScalableTextContainer>
        </Text>
      )}
    </View>
    {buttonText && (
      <View {...css({ float: 'right', margin: '20px 15px 0px' })}>
        <Button onClick={onButtonClick}>{buttonText}</Button>
      </View>
    )}
  </View>
)

export default BigTitleBar
