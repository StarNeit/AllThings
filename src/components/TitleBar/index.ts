export { default as CustomTitleBar } from './CustomTitleBar'
export { default as TitleBarBackButton } from './TitleBarBackButton'
export { default as TitleBarMyFlatButton } from './TitleBarMyFlatButton'
export { default as TitleBarButton } from './TitleBarButton'
export { default as TitleBarButtonGroup } from './TitleBarButtonGroup'
export { default as TitleBarLabel } from './TitleBarLabel'
