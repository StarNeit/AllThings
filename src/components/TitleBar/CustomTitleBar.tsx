import React from 'react'
import { TitleBar } from '@allthings/elements'
import { ITitleBarProps } from '@allthings/elements/TitleBar'
import { useTheme } from '@allthings/elements/Theme'

interface IProps extends ITitleBarProps {
  children?: React.ReactNode
}

const CustomTitleBar = ({ children, ...props }: IProps) => {
  const { theme } = useTheme()
  return (
    <TitleBar {...props} color={theme.background}>
      {children}
    </TitleBar>
  )
}

export default CustomTitleBar
