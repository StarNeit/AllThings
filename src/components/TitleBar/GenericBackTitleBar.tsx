import React, { MouseEvent, ReactNode } from 'react'
import { CustomTitleBar } from 'components/TitleBar/index'
import { SquareIconButton, Text, View } from '@allthings/elements'
import { FormattedMessage } from 'react-intl'

export interface IProps {
  onBack: (event: MouseEvent) => void
  backText?: ReactNode
}

const DefaultBackText = (
  <FormattedMessage
    id="title-bar.generic-back-button"
    description="Generic title bar back button label"
    defaultMessage="Back"
  />
)

const GenericBackTitleBar = ({
  onBack,
  backText = DefaultBackText,
  ...props
}: IProps) => (
  <CustomTitleBar {...props}>
    <View
      data-e2e="back-button"
      direction="row"
      onClick={onBack}
      alignV="center"
    >
      <SquareIconButton icon="arrow-left-filled" iconSize="xs" />
      <Text weight="semi-bold">{backText}</Text>
    </View>
  </CustomTitleBar>
)

export default GenericBackTitleBar
