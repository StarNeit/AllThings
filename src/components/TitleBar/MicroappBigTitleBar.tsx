import React, { ReactNode } from 'react'
import { MicroApps } from '../../enums'
import { useSelector } from 'react-redux'
import BigTitleBar, { IBigTitleBarProps } from 'components/TitleBar/BigTitleBar'
import { getLocalized } from 'containers/Localized'
import { navigateToMyFlat } from 'utils/sendNativeEvent'

interface IProps extends Omit<IBigTitleBarProps, 'title'> {
  subTitle?: ReactNode
  type: MicroApps
}

export const appLabelSelector = (type: MicroApps) => (state: IReduxState) =>
  getLocalized(
    state.app.microApps.find(app => app.type === type).label,
    state.app.locale,
  )

const MicroappBigTitleBar = ({ type, ...props }: IProps) => {
  const microAppLabel = useSelector(appLabelSelector(type))
  const isEmbeddedLayout = useSelector(
    (state: IReduxState) => state.app.embeddedLayout,
  )
  const accessToken = useSelector(
    (state: IReduxState) => state.authentication.accessToken,
  )

  const showBackButton =
    isEmbeddedLayout &&
    type !== MicroApps.PINBOARD &&
    type !== MicroApps.HELPDESK

  return (
    <BigTitleBar
      title={microAppLabel}
      showBackButton={showBackButton}
      onBackButtonClick={() => navigateToMyFlat(accessToken)}
      {...props}
    />
  )
}

export default MicroappBigTitleBar
