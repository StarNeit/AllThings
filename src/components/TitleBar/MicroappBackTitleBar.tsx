import React, { MouseEvent } from 'react'
import GenericBackTitleBar from './GenericBackTitleBar'
import { MicroApps } from '../../enums'
import { useSelector } from 'react-redux'
import { appLabelSelector } from 'components/TitleBar/MicroappBigTitleBar'

interface IProps {
  onBack: (event: MouseEvent) => void
  type: MicroApps
}

const MicroappBackTitleBar = ({ onBack, type }: IProps) => {
  const microAppLabel = useSelector(appLabelSelector(type))

  return <GenericBackTitleBar onBack={onBack} backText={microAppLabel} />
}

export default MicroappBackTitleBar
