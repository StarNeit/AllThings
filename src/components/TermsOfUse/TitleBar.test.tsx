import React from 'react'
import { TitleBar } from './TitleBar'

describe('Check the TitleBar component of the terms of use', () => {
  it('should render correctly for logged in users', () => {
    const mockClick = jest.fn()
    const wrapper = global.mount(
      <TitleBar loggedIn={true} onClick={mockClick} title="A nice title" />,
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.simulate('click')
    expect(mockClick.mock.calls.length).toBe(0)
  })
  it('should render correctly for non logged in users and handle going back', () => {
    const mockClick = jest.fn()
    const wrapper = global.mount(
      <TitleBar onClick={mockClick} title="A nice title" />,
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.simulate('click')
    expect(mockClick.mock.calls.length).toBe(1)
  })
})
