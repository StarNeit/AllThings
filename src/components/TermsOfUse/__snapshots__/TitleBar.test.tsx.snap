// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the TitleBar component of the terms of use should render correctly for logged in users 1`] = `
.css-0,
[data-css-0] {
  background: #1cabbb;
  border-radius: 0;
  cursor: auto;
  height: 50px;
}

.css-1,
[data-css-1] {
  margin: 10px;
  padding-top: 0px;
}

.css-2,
[data-css-2] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-flex: 0 0 auto;
}

.css-3,
[data-css-3] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 600;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #ffffff;
}

.css-4,
[data-css-4] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

<ForwardRef
  alignH="space-between"
  alignV="center"
  data-css-0=""
  direction="row"
  onClick={[Function]}
>
  <div
    data-css-2=""
    data-css-0=""
    onClick={[Function]}
  >
    <ForwardRef
      data-css-3=""
      data-css-1=""
    >
      <div
        data-css-3=""
        data-css-1=""
        data-css-4=""
      >
        A nice title
      </div>
    </ForwardRef>
  </div>
</ForwardRef>
`;

exports[`Check the TitleBar component of the terms of use should render correctly for non logged in users and handle going back 1`] = `
.css-0,
[data-css-0] {
  margin: 10px;
  padding-top: 0px;
}

.css-1,
[data-css-1] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-flex: 0 0 auto;
}

.css-2,
[data-css-2] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 600;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #ffffff;
}

.css-3,
[data-css-3] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-4,
[data-css-4] {
  background: #1cabbb;
  border-radius: 5px 5px 0 0;
  cursor: pointer;
  height: 40px;
}

.css-5,
[data-css-5] {
  width: 16px;
  height: 16px;
  fill: #ffffff;
}

<ForwardRef
  alignH="space-between"
  alignV="center"
  data-css-4=""
  direction="row"
  onClick={[MockFunction]}
>
  <div
    data-css-1=""
    data-css-4=""
    onClick={[MockFunction]}
  >
    <ForwardRef
      data-css-2=""
      data-css-0=""
    >
      <div
        data-css-2=""
        data-css-0=""
        data-css-3=""
      >
        A nice title
      </div>
    </ForwardRef>
    <ForwardRef
      alignH="center"
      alignV="center"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      data-css-5=""
      data-css-0=""
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        data-css-5=""
        data-css-0=""
        data-css-3=""
      />
    </ForwardRef>
  </div>
</ForwardRef>
`;
